### PR TITLE
Update rubocop-performance gem and enable Performance/DeletePrefix and Performance/DeleteSuffix cops

### PR DESCRIPTION
**Pull request by fatkodima**

### CODE REVIEW

File: actionview/lib/action_view/helpers/form_helper.rb [hunk 1]
```diff
@@ -1668,7 +1668,7 @@ def initialize(object_name, object, template, options)
 
         convert_to_legacy_options(@options)
 
-        if @object_name.to_s.match(/\[\]$/)
+        if @object_name.to_s.match(/\[\]$/) # rubocop:disable Performance/EndWith
```
Comment (kamipo) [inline actionview/lib/action_view/helpers/form_helper.rb:1671]: Why use `rubocop:disable`?

Comment (fatkodima) [reply actionview/lib/action_view/helpers/form_helper.rb:1671]: delete_suffix cannot be used - matchdata is used in the next line (`Regexp.last_match.pre_match`)

Comment (kamipo) [reply actionview/lib/action_view/helpers/form_helper.rb:1671]: Use `if foo.end_with("[]"); foo.delete_suffix("[]")` as a faster alternative.

Comment (fatkodima) [reply actionview/lib/action_view/helpers/form_helper.rb]: ok 👍 
Updated, and in other places. CI failures seems not related to this PR.

**Pull request merged**